src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -87,6 +87,8 @@ interface BuildConfiguration {
 
     fun fetchBuildTags(): List<String>
 
+    fun fetchBuildArtifactDependencies(): List<ArtifactDependency>
```

Comment:
No, you shouldn't, firstly because this would break backward compatibility of API. If you really want to rename something you need to leave the old function as deprecated for some time to simplify migration.

Anyway this interface represents a build configuration and if we would name a method 'fetchTags' one may think that it fetches tags of the build configuration. But it fetches tags from all builds from this configuration, so the name 'fetchBuildTags' is the right choice here.

But 'fetchBuildArtifactDependencies' actually fetches artifact dependencies of the build configuration itself. So 'build' word here not just makes the name longer, but also may be misleading: one may think that this method processes all builds of this configuration and fetches artifact dependencies from all of them.